wremnants/muon_selections.py

```diff
@@ -56,12 +56,15 @@ def select_veto_muons(df, nMuons=1, condition="==", ptCut=10.0, etaCut=2.4):
 
 '''
 
-def select_veto_muons(df, nMuons=1, condition="==", ptCut=15.0, etaCut=2.4):
+def select_veto_muons(df, nMuons=1, condition="==", ptCut=15.0, etaCut=2.4, useGlobalOrTrackerVeto=False):
 
     # n.b. charge = -99 is a placeholder for invalid track refit/corrections (mostly just from tracks below
     # the pt threshold of 8 GeV in the nano production)
     df = df.Define("vetoMuonsPre", "Muon_looseId && abs(Muon_dxybs) < 0.05 && Muon_correctedCharge != -99")
-    df = df.Define("vetoMuonsPre2", "vetoMuonsPre && Muon_isGlobal && Muon_highPurity && Muon_standalonePt > 15 && Muon_standaloneNumberOfValidHits > 0 && wrem::vectDeltaR2(Muon_standaloneEta, Muon_standalonePhi, Muon_correctedEta, Muon_correctedPhi) < 0.09")
+    if not useGlobalOrTrackerVeto:
+        df = df.Define("vetoMuonsPre2", "vetoMuonsPre && Muon_isGlobal && Muon_highPurity && Muon_standalonePt > 15 && Muon_standaloneNumberOfValidHits > 0 && wrem::vectDeltaR2(Muon_standaloneEta, Muon_standalonePhi, Muon_correctedEta, Muon_correctedPhi) < 0.09")
+    else:
+        df = df.Define("vetoMuonsPre2", "vetoMuonsPre && ((Muon_isGlobal && Muon_standalonePt > 15 && Muon_highPurity && Muon_standaloneNumberOfValidHits > 0 && wrem::vectDeltaR2(Muon_standaloneEta, Muon_standalonePhi, Muon_correctedEta, Muon_correctedPhi) < 0.09) || (Muon_highPurity && Muon_isTracker && Muon_innerTrackOriginalAlgo != 13 && Muon_innerTrackOriginalAlgo != 14))")
```

Comment:
For readability would probably be better to define some common strings for the global and tracker pieces of this